Makefile.am

```diff
@@ -915,6 +915,36 @@ php_EXTRA_DIST=                                                       \
 
 python_EXTRA_DIST=                                                           \
   python/MANIFEST.in                                                         \
+  python/generate_docs.py                                                    \
```

Comment:
Move those files to nodist_... (maybe nodist_python_doc) instead of python_EXTRA_DIST. Because we do not want to add these files into release package.

Examples of other not released files that use nodist_
https://github.com/protocolbuffers/protobuf/search?q=nodist&unscoped_q=nodist

Or I am wondering if it is better to move this tool to google3 (google3/third_party/protobuf/testing/protobuf/python/)?

Reply:
I tried adding:

```

nodist_python_docs_SOURCES=                                                  \
  python/generate_docs.py                                                    \
  python/docs/requirements.txt                                               \
  python/docs/make.bat                                                       \
  python/docs/index.rst                                                      \
  python/docs/google/protobuf/wrappers_pb2.rst                               \
  python/docs/google/protobuf/type_pb2.rst                                   \
  python/docs/google/protobuf/timestamp_pb2.rst                              \
  python/docs/google/protobuf/text_format.rst                                \
  python/docs/google/protobuf/text_encoding.rst                              \
  python/docs/google/protobuf/symbol_database.rst                            \
  python/docs/google/protobuf/struct_pb2.rst                                 \
  python/docs/google/protobuf/service_reflection.rst                         \
  python/docs/google/protobuf/service.rst                                    \
  python/docs/google/protobuf/reflection.rst                                 \
  python/docs/google/protobuf/proto_builder.rst                              \
  python/docs/google/protobuf/message_factory.rst                            \
  python/docs/google/protobuf/message.rst                                    \
  python/docs/google/protobuf/json_format.rst                                \
  python/docs/google/protobuf/field_mask_pb2.rst                             \
  python/docs/google/protobuf/empty_pb2.rst                                  \
  python/docs/google/protobuf/duration_pb2.rst                               \
  python/docs/google/protobuf/descriptor_pool.rst                            \
  python/docs/google/protobuf/descriptor_pb2.rst                             \
  python/docs/google/protobuf/descriptor_database.rst                        \
  python/docs/google/protobuf/descriptor.rst                                 \
  python/docs/google/protobuf/any_pb2.rst                                    \
  python/docs/google/protobuf.rst                                            \
  python/docs/environment.yml                                                \
  python/docs/conf.py                                                        \
  python/docs/Makefile
```

but it has no effect on `build_cpp_distcheck`. Instead, in 30763ff0e37f8f5329a88884af9de1ff15313c16 I've updated `tests.sh` to exclude the docs directory and moved the `generate_docs.py` script there.

> Or I am wondering if it is better to move this tool to google3 (google3/third_party/protobuf/testing/protobuf/python/)?

Which tool?

Move the sphinx config to internal? Most open source packages provide a way to build the docs for the package. Having the docs here would encourage contributions to the docs, which can be an easier way to get started with contributing to a project than diving into the code.

Or move the `python_EXTRA_DIST` stuff? It wasn't documented in the CONTRIBUTING file, but after a bit of digging, I was able to figure out how that I could install `libtool` and `automake` and run `./tests.sh cpp_distcheck`.

I had to switch to Python 2 due to

```
"./scripts/fuse_gmock_files.py" "./fused-src"
  File "./scripts/fuse_gmock_files.py", line 235
    print __doc__
                ^
SyntaxError: Missing parentheses in call to 'print'. Did you mean print(__doc__)?
```

Reply:
Yes I was wondering Move the sphinx config to internal. But it sounds reasonable to be here as you mentioned it would encourage contributions